src/vbe_info.rs

```diff
@@ -49,35 +94,91 @@ impl fmt::Debug for VBEControlInfo {
     }
 }
 
+
+/// Extended information about a specific VBE display mode from the
+/// mode list returned by `VBEControlInfo` (VBE Function `00h`).
 #[derive(Copy, Clone)]
 #[repr(C, packed)]
 pub struct VBEModeInfo {
+    /// Mode attributes.
     pub mode_attributes: VBEModeAttributes,
+
+    /// Window A attributes.
     pub window_a_attributes: VBEWindowAttributes,
+
+    /// Window B attributes.
     pub window_b_attributes: VBEWindowAttributes,
-    pub window_granularity: u16, // Measured in kilobytes.
+
+    /// Window granularity (Measured in Kilobytes.)
+    pub window_granularity: u16,
+
+    /// Window size.
     pub window_size: u16,
+
+    /// Window A start segment.
     pub window_a_segment: u16,
+
+    /// Window B start segment.
     pub window_b_segment: u16,
+
+    /// Real mode pointer to window function.
     pub window_function_ptr: u32,
+
+    /// Bytes per scan line
     pub pitch: u16,
+
+    /// Horizontal and vertical resolution in pixels or characters.
     pub resolution: (u16, u16),
+
+    /// Character cell width and height in pixels.
     pub character_size: (u8, u8),
+
+    /// Number of memory planes.
     pub number_of_planes: u8,
+
+    /// Bits per pixel
     pub bpp: u8,
+
+    /// Number of banks
     pub number_of_banks: u8,
+
+    /// Memory model type
     pub memory_model: VBEMemoryModel,
-    pub bank_size: u8, // Measured in kilobytes.
+
+    /// Bank size (Measured in Kilobytes.)
+    pub bank_size: u8,
+
+    /// Number of images.
     pub number_of_image_pages: u8,
+
+    /// Reserved for page function.
     reserved0: u8,
+
+    /// Red colour field.
     pub red_field: VBEField,
+
+    /// Green colour field.
     pub green_field: VBEField,
+
+    /// Blue colour field.
     pub blue_field: VBEField,
+
+    /// Reserved colour field.
     pub reserved_field: VBEField,
+
+    /// Direct colour mode attributes.
     pub direct_color_attributes: VBEDirectColorAttributes,
+
+    /// Physical address for flat memory frame buffer
     pub framebuffer_base_ptr: u32,
+
+    /// This should be a reserved field???
```

Comment:
I'm using [this pdf](https://github.com/mental32/osdev/blob/master/vbe3.pdf) as the refrence for VBE 3.0 and the relevant part of the struct can be found on page 31.

The section thats capped with `; Mandatory information for VBE 2.0 and above` lists the base address which is double word (32 bits, 4 bytes) and two more double worlds that are marked as reserved.

I'm not quite sure what to do here, since I don't have the relevant reference to document these fields.

Reply:
It seems like this is a change from VBE 2.0 specification to VBE 3.0 specification.  If we want to support those fields, we need to support handling the differences between VBE 2.0 and VBE 3.0.  My gut reaction is to mark these fields as deprecated to encourage following the 3.0 specification.  At the very least, we should include a note that the VBE 3.0 specification declares these fields should be zero.

@phil-opp @IsaacWoods  Any thoughts?